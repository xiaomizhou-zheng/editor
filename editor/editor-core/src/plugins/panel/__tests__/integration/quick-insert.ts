import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  editable,
  expectToMatchDocument,
  fullpage,
  quickInsert,
} from '@atlaskit/editor-test-helpers/integration/helpers';
import {
  goToEditorTestingWDExample,
  mountEditor,
} from '@atlaskit/editor-test-helpers/testing-example-page';
import { selectors } from './_utils';

BrowserTestCase(
  'quick-insert.ts: Insert panel via quick insert',
  {},
  async (client: any, testName: string) => {
    const page = await goToEditorTestingWDExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
      allowPanel: true,
    });

    await page.click(fullpage.placeholder);
    await quickInsert(page, 'Info panel');
    await page.waitForSelector(selectors.PANEL_EDITOR_CONTAINER);

    await page.type(editable, 'this text should be in the panel');

    await expectToMatchDocument(page, testName);
  },
);

BrowserTestCase(
  'quick-insert.ts: Insert custom panel via quick insert',
  {},
  async (client: any, testName: string) => {
    const page = await goToEditorTestingWDExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
      allowPanel: { allowCustomPanel: true, allowCustomPanelEdit: true },
    });

    await page.click(fullpage.placeholder);
    await quickInsert(page, 'Custom panel');
    await page.waitForSelector(selectors.PANEL_EDITOR_CONTAINER);

    await page.type(editable, 'this text should be in the panel');

    await expectToMatchDocument(page, testName);
  },
);
