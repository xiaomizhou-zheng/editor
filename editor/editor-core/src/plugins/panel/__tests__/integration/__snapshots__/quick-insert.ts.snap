// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quick-insert.ts: Insert custom panel via quick insert 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelColor": "#E6FCFF",
        "panelIcon": ":rainbow:",
        "panelIconId": "1f308",
        "panelIconText": "ðŸŒˆ",
        "panelType": "custom",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "this text should be in the panel",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`quick-insert.ts: Insert panel via quick insert 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "this text should be in the panel",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;
